src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -381,12 +381,18 @@ public void setName(String name) {
      * @return interned string
      */
     protected String getCachedString(String str) {
-        WeakReference<String> internedString = stringCache.get(str);
-        if ( internedString == null || internedString.get() == null) {
-            internedString = new WeakReference<>(str);
-            stringCache.put(str, internedString);
+        if (!vcfIndexUseStringsCache)
+            return str;
+        WeakReference<String> stringWeakReference = stringCache.get(str);
```

Comment:
re 1 - note the map is a WeakHashMap - to my understanding that will solve the issue for the keys as well, not just the values, no?
re 2 - indeed, but the flip side is that maintaining an LFU (LFUDA) isn't free either. 
I think this whole things pivots around one basic thing: benchmarks. without conducting some, preferably on varied data-sets, I can't see how one can "apriori" conclude which is preferable: no-caching / weak references / interning / LRU's etc.